snoop/data/tasks.py

```diff
@@ -284,21 +283,30 @@ def dispatch_pending_tasks():
         ])
     )
 
-    for task in task_query:
+    task_count = task_query.count()
```

Comment:
In Postgres this is [surprisingly expensive](https://wiki.postgresql.org/wiki/Slow_Counting) if there are many rows in the count. It might be a better idea to query for a single item and see if you receive anything.